tools/image_processing/yolo/json2yolosegment.xml

```diff
@@ -0,0 +1,49 @@
+<tool id="json2yolosegment" name="Convert AnyLabeling JSON to YOLO text" version="@VERSION@+galaxy0" profile="24.2">
+    <description>with ultralytics</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="creator" />
+    <command detect_errors="aggressive">
+    <![CDATA[
+      mkdir ./input ./output &&
+		    
+      #for $i, $filename in enumerate($in_json):
+          ln -s '$filename' './input/${filename.element_identifier}' &&
+      #end for
+      python '$__tool_directory__/json2yolosegment.py' -i ./input/ -o ./output -c '$class_name'
+		
+    ]]>
+    </command>
+    <inputs>
+	 <param name="in_json" type="data" format="json" multiple="true" label="Input label files" help="The label file is the output of AnyLabeling in JSON format"/>
+         <param name="class_name" type="data" format="txt" label="Class file" help="Class names text file, contains the names of classes, one class per row"/>
+    </inputs>
+    <outputs>
+        <collection name="output_yolo" type="list" label="YOLO Text File">
+		<discover_datasets pattern="(?P&lt;name&gt;.+)\.txt" visible="true" directory="./output" format="txt"/>
+        </collection>
+    </outputs>
+    <tests>
+        <test>
+            <param name="in_json" value="in_json.json,in_json1.json" />
+            <param name="class_name" value="class_names.txt" />
+            <output_collection name="output_yolo">
+                <element name="in_json">
+                    <assert_contents><has_size value="1819" /></assert_contents>
```

Comment:
please do not use `size`. This is not a good test. Try to assert the content. How many lines, rows, text snippets that are included ...